src/averageAges.js

```diff
@@ -15,11 +15,13 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => century
+    ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+    : person.sex === 'm'
+  );
+  const menLife = men.map(person => person.died - person.born);
+
+  return menLife.reduce((sum, age) => sum + age, 0) / menLife.length;
```

Comment:
Move this logic to separate function so you will be able to reuse it

Reply:
tried to do it